src/main/groovy/com/deploygate/gradle/plugins/DeployGatePlugin.groovy

```diff
@@ -54,108 +55,88 @@ class DeployGatePlugin implements Plugin<Project> {
-                task.dependsOn(DeployGateTaskFactory.uploadAabTaskName(d.name))
+            project.tasks.named(Constants.SUFFIX_AAB_TASK_NAME).configure { task ->
+                task.dependsOn(Constants.uploadAabTaskName(deployment.name))
             }
 
-            project.tasks.register(DeployGateTaskFactory.uploadApkTaskName(d.name), UploadApkTask) { task ->
-                final NamedDeployment deployment = project.deploygate.findDeploymentByName(d.name)
-
+            project.tasks.register(Constants.uploadApkTaskName(deployment.name), UploadApkTask) { task ->
                 if (!deployment.skipAssemble) {
-                    task.logger.debug("${d.name} required assmble but ignored")
+                    task.logger.debug("${deployment.name} required assmble but ignored")
                 }
 
-                task.variantName = d.name
-                task.dependsOn(Constants.LOGIN_TASK_NAME)
-
-                task.configuration = UploadApkTask.createConfiguration(deployment, new DefaultPresetApkInfo(d.name))
-                task.applyTaskProfile()
+                task.credentials.set(loginTask.map { it.credentials })
+                task.deployment.copyFrom(deployment)
+                task.apkInfo.set(new DefaultPresetApkInfo(deployment.name))
+                task.dependsOn(loginTask)
             }
 
-            project.tasks.register(DeployGateTaskFactory.uploadAabTaskName(d.name), UploadAabTask) { task ->
-                final NamedDeployment deployment = project.deploygate.findDeploymentByName(d.name)
-
+            project.tasks.register(Constants.uploadAabTaskName(deployment.name), UploadAabTask) { task ->
                 if (!deployment.skipAssemble) {
-                    task.logger.debug("${d.name} required assmble but ignored")
+                    task.logger.debug("${deployment.name} required assmble but ignored")
                 }
 
-                task.variantName = d.name
-                task.dependsOn(Constants.LOGIN_TASK_NAME)
-
-                task.configuration = UploadAabTask.createConfiguration(deployment, new DefaultPresetAabInfo(d.name))
-                task.applyTaskProfile()
+                task.credentials.set(loginTask.map { it.credentials })
+                task.deployment.copyFrom(deployment)
+                task.aabInfo.set(new DefaultPresetAabInfo(deployment.name))
+                task.dependsOn(loginTask)
             }
         }
 
         AndroidGradlePlugin.ifPresent(project) {
             project.android.applicationVariants.configureEach { /* ApplicationVariant */ variant ->
                 def variantProxy = new IApplicationVariantImpl(variant)
 
-                namedOrRegister(project, DeployGateTaskFactory.uploadApkTaskName(variantProxy.name), UploadApkTask).configure { task ->
-                    final NamedDeployment deployment = project.deploygate.findDeploymentByName(variantProxy.name)
```

Comment:
After introducing DeploymentConfiguration, don't need to touch NamedDeployment anymore in AGP domain because `project.deploygate.deployments.configureEach` already set values if it exists.